### DISCUSSION TITLE
Dask Distributed: Limit Dask distributed worker to 1 CPU

### DISCUSSION THREAD

**Discussion by smisra1** [dask/distributed#7649] [Q&A]
My system has 4 CPU, 16 GB RAM. My Aim is to deploy dask distributed workers that use 1 CPU each ONLY to run code assigned to them.

I am deploying a scheduler container and worker containers using docker to run a code that uses Dask delayed and dask dataframes.

Following are the Docker run commands for both:

Scheduler
```
docker run --name dask-scheduler --network host daskdev/dask:2023.1.1-py3.10 dask-scheduler
```
worker (multiple docker run commands tried, all different combinations of following command)
```
docker run --name dask-worker --network host dask-worker --nworkers 1 --nthreads 1 --resources "process=1" --memory-limit 2 GiB --name dask-worker daskdev/dask:2023.1.1-py3.10 10.76.8.50:8786
```
Now what I wanted was that when dask.compute(scheduler="processes") is run, the worker will use only 1 cpu for running the code. However, atleast 3 CPU can be seen at 100% capacity.

![cb2zY](https://user-images.githubusercontent.com/83804524/224890036-ee985095-f008-444e-b5e5-a5cc1e5cbc9c.png)

Is there something I have missed?
How could I limit a distributed worker to use ONLY 1 CPU for this compute work?

**Comment by fjetter** [comment]
We typically organize help around usage related questions on [discourse](https://dask.discourse.group/). Could you please repost your question there?
